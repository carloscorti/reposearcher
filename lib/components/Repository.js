import React from 'react';
import PropTypes from 'prop-types';

/**
 * Repository comoponent: Displays repo card
 *
 * @param  {Object}  props  must have "data" key, and data object must have
 *   name, description, language, updated_at, html_url keys
 */

const Repository = ({ data }) => {
  /**
   * Data Object: contains repo card info to display
   *
   * @key  {string}  name         repo short name
   * @key  {string}  description  repo description
   * @key  {string}  language     repo main programming lenguaje
   * @key  {string}  updated_at   repo last updated date
   * @key  {string}  html_url     repo url
   */
  const { name, description, language, updated_at, html_url } = data;

  const updatedDate = new Date(updated_at);
  const daysSinceUpdate = Math.round(
    (Date.now() - updatedDate) / (1000 * 60 * 60 * 24)
  );

  return (
    <>
      <div className="pinned-repo-wrapper">
        <div className="pinned-repo">
          <h3>{name}</h3>
          <p>{description}</p>
          <ul className="stats">
            <li>
              <span className="icon">ðŸ”µ</span>
              <span>{language}</span>
            </li>
            <li>
              <span className="icon"></span>
              <span>Updated {daysSinceUpdate} days ago</span>
            </li>
            <li>
              <span className="icon"></span>
              <span>
                <a href={html_url} target="_blank" rel="noopener noreferrer">
                  Go to repo
                </a>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </>
  );
};

Repository.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    language: PropTypes.string,
    updated_at: PropTypes.string,
    html_url: PropTypes.string,
  }).isRequired,
};

export default Repository;
