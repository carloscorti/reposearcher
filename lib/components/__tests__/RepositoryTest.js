import React from 'react';
import Repository from '../Repository';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('Repository', () => {
  const testProps = {
    repo: {
      id: 1,
      data: {
        name: 'n1',
        description: 'd1',
        language: 'l1',
        updated_at: '2020-04-16T20:33:43Z',
        html_url: 'url1',
      },
    },
  };
  it('renders correctly', () => {
    const wrapper = shallow(
      <Repository key={testProps.repo.id} data={testProps.repo} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
