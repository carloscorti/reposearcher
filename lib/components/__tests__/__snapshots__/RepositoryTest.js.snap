// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository renders correctly 1`] = `
<Fragment>
  <div
    className="pinned-repo-wrapper"
  >
    <div
      className="pinned-repo"
    >
      <h3 />
      <p />
      <ul
        className="stats"
      >
        <li>
          <span
            className="icon"
          >
            ðŸ”µ
          </span>
          <span />
        </li>
        <li>
          <span
            className="icon"
          />
          <span>
            Updated 
            NaN
             days ago
          </span>
        </li>
        <li>
          <span
            className="icon"
          />
          <span>
            <a
              rel="noopener noreferrer"
              target="_blank"
            >
              Go to repo
            </a>
          </span>
        </li>
      </ul>
    </div>
  </div>
</Fragment>
`;
